### ISSUE TITLE
LCP: correct a crash on loading revoked licenses

### ISSUE THREAD

**Issue by llemeurfr** [readium/r2-testapp-swift#294]
We're witnessing weird crashes and issues with revoked licenses. Reproductible with licenses from the sandbox server, attached as zip: 

[revoked-licenses.zip](https://github.com/readium/r2-testapp-swift/files/3709328/revoked-licenses.zip)

dante-hell-revoked.lcpl -> the content is downloaded and the book opens while it should not (the status doc contains de revoked status). 

wasteland-revoked.lcpl -> the app can crash or show a message with corrupted data ("on" can also be followed by a large integer).

![IMG_1213](https://user-images.githubusercontent.com/14943186/66518731-2bd0e980-eae6-11e9-84c7-34acf09c4f7d.png)

**Comment by mickael-menu** [comment]
Regarding `dante-hell-revoked.lcpl`, this is caused by the fact that the license is not available anymore on the server (404 at https://front-prod.edrlab.org/api/v1/licenses/7ac7befa-fe83-4e33-a087-44c5f91ff39b). After fetching the LSD, the app notices that a new version of the License is available and tries to fetch it. Since it fails, the workflow continues with the local License, and since it is not expired (end date valid), then the book is opened.

This is despite the status in the LSD, because according to the spec: "If the status value of a Status Document contradicts the corresponding up-to-date License Document, the License Document takes precedence."

There is an ambiguity with "corresponding up-to-date License Document". Since we can't download the updated License Document, then we rely on the outdated one. Maybe we should make an exception in case the License Document can't be updated, and make the LSD take precedence over the outdated License Document in this case?

So I'm not sure if it's a server-wise error, or a rule that is not fail-safe on the client side.

**Comment by llemeurfr** [comment]
I checked quickly the lcp server via its api and the license is there. The record of the purchase seems to have disappeared from the test frontend server, which surprises me. But this is a server side issue. 

For our issue, the license was issued on 2019-01-17T14:57:38Z. It was updated on 2019-01-17T17:10:54Z. So you're right, the app must try to get the updated license. 

But I think, as you are proposing, that as the updated license is not available, the status value is king (there is no contradiction, there is one piece missing which is the updated license). 

I updated https://github.com/readium/architecture/tree/master/other/lcp section 3 and 4 in this direction.
